src/test/java/com/thealgorithms/datastructures/trees/SplayTreeTest.java

```diff
@@ -0,0 +1,110 @@
+package com.thealgorithms.datastructures.trees;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class SplayTreeTest {
+
+    @ParameterizedTest
+    @MethodSource("traversalStrategies")
+    public void testTraversal(SplayTree.TreeTraversal traversal, List<Integer> expected) {
+        SplayTree tree = createComplexTree();
+        List<Integer> result = tree.traverse(traversal);
+        assertEquals(expected, result);
+    }
```

Comment:
In order to have the _100%_ coverage, I think that this this test should be also parametrized over the `tree`. Could you please rewrite this test?

Reply:
Sure, I will do it

Reply:
If necessary, I'll open another pull request to achieve 100% code coverage. I don't have much time to do it in this PR. Thanks for your suggestion @vil02. Can we merge this PR into the repo when the coverage is not 100%?

Reply:
> Can we merge this PR into the repo when the coverage is not 100%?

No.